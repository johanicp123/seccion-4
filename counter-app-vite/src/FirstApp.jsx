export function App(){
    return (<h1>Hola mundo1111</h1>);
}


