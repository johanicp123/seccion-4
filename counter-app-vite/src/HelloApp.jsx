import React from 'react'

export const Nombre = () => {
  return ( <h1>hola desde funciom</h1> );
}



const newMessage = {
  message: 'Hola mundo',
  title: 'Alberth'
};

export const HelloApp = () => {
  return (
    <>
        <code> {JSON.stringify(newMessage)}</code>
        <h1>{Nombre()}</h1>
    <p>holaaaa</p>
    </>
  )
}
