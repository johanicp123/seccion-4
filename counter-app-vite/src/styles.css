html,body{
    background-color: black;
    color: white;
    font-family: 'Times New Roman', Times, serif;
    font-size: 1.3rem;
    padding: 70px;
}