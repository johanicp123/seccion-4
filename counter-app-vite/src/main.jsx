import React from "react";
import  ReactDOM from "react-dom/client";
import { CounterApp } from "./CounterApp";

//import {FirstApp} from './FirstApp';
//import { HelloApp } from "./HelloApp";
import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={20}/>
    </React.StrictMode>
);
