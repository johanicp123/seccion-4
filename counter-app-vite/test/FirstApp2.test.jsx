/** @jest-environment jsdom */
import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";
describe("prueba en FirstApp", () => {
  const title = "Hola, soy goku";
  const subtitle = "soy un subtitulo";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar el mensaje hola soy goku", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe mostrar el tituto en un h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("debe mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subTitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
