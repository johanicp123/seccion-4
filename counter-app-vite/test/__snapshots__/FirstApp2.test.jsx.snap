// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en FirstApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy goku
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay subtitulo
  </p>
  <p>
    Alberth Campos
  </p>
</div>
`;
